prow/test/integration/test/gerrit_test.go

```diff
@@ -34,71 +35,128 @@ const (
+		name     string
+		change   gerrit.ChangeInfo
+		messages []string
+	}{
+		{
+			name: "1 New change with 1 presubit triggered",
+			change: gerrit.ChangeInfo{
+				CurrentRevision: "1",
+				ID:              "1",
+				ChangeID:        "1",
+				Project:         "gerrit-test-infra",
+				Branch:          "master",
+				Status:          "NEW",
+				Revisions:       map[string]client.RevisionInfo{"1": {Number: 1, Ref: "refs/changes/00/1/1"}},
+				Messages:        []gerrit.ChangeMessageInfo{{RevisionNumber: 1, Message: "/test all", ID: "1"}},
+			},
+			messages: []string{"/test all", "Triggered 1 prow jobs (0 suppressed reporting): \n  * Name: hello-world-presubmit"},
+		},
+		{
+			name: "no presubmit Prow jobs automatically triggered from WorkInProgess change",
+			change: client.ChangeInfo{
+				CurrentRevision: "1",
+				ChangeID:        "2",
+				ID:              "2",
+				Project:         "gerrit-test-infra",
+				Status:          "NEW",
+				Branch:          "master",
+				WorkInProgress:  true,
+				Revisions: map[string]gerrit.RevisionInfo{
+					"1": {
+						Number: 1001,
+					},
+				},
+			},
+			messages: []string{},
+		},
+		{
+			name: "presubmit runs when a file matches run_if_changed",
+			change: client.ChangeInfo{
+				CurrentRevision: "1",
+				ChangeID:        "3",
+				ID:              "3",
+				Branch:          "master",
+				Project:         "gerrit-test-infra",
+				Status:          "NEW",
+				Messages:        []gerrit.ChangeMessageInfo{{RevisionNumber: 1, Message: "/test all", ID: "1"}},
+				Revisions: map[string]client.RevisionInfo{
+					"1": {
+						Number: 1001,
+						Files: map[string]client.FileInfo{
+							"bee-movie-script.txt": {},
+							"africa-lyrics.txt":    {},
+							"important-code.go":    {},
+						},
+					},
+				},
+			},
+			messages: []string{"/test all", "Triggered 2 prow jobs (0 suppressed reporting): \n  * Name: hello-world-presubmit\n  * Name: bee-movie-presubmit"},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+
+			gerritClient, err := client.NewClient(map[string][]string{gerritServer: {"fakegerritserver"}})
+			if err != nil {
+				t.Fatalf("Failed creating gerritClient: %v", err)
+			}
+
+			account := gerrit.AccountInfo{
+				AccountID: 1,
+				Name:      "Prow Bot",
+				Username:  "testbot",
+			}
+
+			branch := gerrit.BranchInfo{}
+
+			if err = addBranchToServer(branch, "gerrit-test-infra", "master"); err != nil {
+				t.Fatalf("failed to add branch to server: %v", err)
+			}
+			if err = addAccountToServer(account); err != nil {
+				t.Fatalf("Failed to add change to server: %s", err)
+			}
+			if err = login(account.AccountID); err != nil {
+				t.Fatalf("Failed to set self on server: %s", err)
+			}
+			if err = addChangeToServer(tc.change, "gerrit-test-infra"); err != nil {
+				t.Fatalf("Failed to add change to server: %s", err)
+
+			}
+
+			//Give some time for gerrit to pick up the change
+			time.Sleep(20 * time.Second)
```

Comment:
Nit: You could use wait.Poll() (https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll) here to wait the minimum number of seconds.